cli/listener/src/main.rs

```diff
@@ -392,7 +403,154 @@ impl Metrics {
     }
 }
 
-fn serve_request(request: Request, metrics_mutex: &MetricsMutex) -> Result<(), std::io::Error> {
+#[derive(Serialize, PartialEq, Debug)]
+struct ResponseError {
+    error: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+struct ResponseInterval {
+    interval_prices: IntervalPrices,
+    annual_percentage_rate: f64,
```

Comment:
```suggestion
    annual_percentage_yield: f64,
```

[According to Investopedia](https://www.investopedia.com/terms/a/apr.asp#toc-apr-vs-annual-percentage-yield-apy) the value we return is an APY, not an APR, because we do take compounding into account.